common/src/API.php

```diff
@@ -17,6 +17,20 @@ public function __construct($api_domain, $shop_id, $api_key) {
         $this->api_key = (string) $api_key;
     }
 
+    public function getReviews(): array {
+        $params = [
+            'id' => $this->shop_id,
+            'code' => $this->api_key,
+            'detailed' => true,
+        ];
+        $url = $this->buildURL('https://' . $this->api_domain . '/api/1.0/product_reviews.xml', $params);
+        $response = Requests::get($url);
+        $response->throw_for_status();
+        $xml =  simplexml_load_string($response->body);
+        $json = json_encode($xml);
+        return json_decode($json,TRUE);
```

Comment:
Also, this will break if there's invalid unicode in any review, which could happen.  I guess you could use `JSON_INVALID_UTF8_SUBSTITUTE` but this is only available on PHP 7.2.0+.  Generally, this just looks like a hack to me.  How difficult is it to use DOMDocument or SimpleXML directly?

Reply:
Initially, I used `SimpleXML`, but because it returns `SimpleXmlObject` I went for an `array`. I was thinking of asking you if SimpleXML would be okay but I did not do it 😐